ccm_web/client/src/pages/Home.tsx

```diff
@@ -30,11 +33,35 @@ const useStyles = makeStyles((theme) => ({
 interface HomeProps extends LtiProps {
   ltiKey: string | undefined
   globals: Globals
+  course: CanvasCourseBase | undefined
+  getCourseError: Error | undefined
+  setCourse: (course: CanvasCourseBase|undefined) => void
 }
 
 function Home (props: HomeProps): JSX.Element {
+  const { enqueueSnackbar } = useSnackbar()
   const classes = useStyles()
 
+  const setCourseNameAsync = async (
+    newCourseName: string
+  ): Promise<CanvasCourseBase|undefined> => {
+    return await apiSetCourseName(props.ltiKey, props.globals.course.id, newCourseName)
+  }
+
+  const [doSetCourseName, setCourseNameLoading, setCourseNameError] = usePromise<CanvasCourseBase|undefined, typeof setCourseNameAsync>(
+    setCourseNameAsync,
+    (course: CanvasCourseBase|undefined) => {
+      props.setCourse(course)
+      enqueueSnackbar('Course name saved', { variant: 'success' })
+    }
+  )
+
+  useEffect(() => {
+    if (setCourseNameError !== undefined) {
+      enqueueSnackbar('Error saving course name', { variant: 'error' })
+    }
```

Comment:
At some point, we will need to think about how validation will work with the snackbar. The `error` object probably contains the messages about why it failed. Could these be exposed to the user, so they know how to respond (i.e. make it shorter, or more than 0 characters)? Should this be in the snackbar or somewhere more permanent? Do we want to add client-side validation? I'm open to different solutions; I think the first might be the simplest for now, but I could see a future requirement for on-type validation, which would mean some kind of FE code. I've done something similar before, so I'm happy to discuss the pros and cons, or share code.

It's okay with me pushing this off to a future issue/PR.

Reply:
I gave it a little thought and wasn't sure how to distinguish between an error that contained a message that we would want to  show the user and one that we didn't.

My thought was that we'd have to return custom errors from the API calls and that would clash with the universal error handling in at least some small way.  If we can figure out a strategy  to handle that I feel like  it's best.  Secondary client side validation seems like an ok idea too, just would have to be manually synced up with the validation rules applied by canvas, and would require some additional enhancement of the InlineTextEdit component I assume.